crates/hir_def/src/type_ref.rs

```diff
@@ -298,3 +304,44 @@ impl TypeBound {
         }
     }
 }
+
+/// A concrete constant value
```

Comment:
when we implement array sizes more 'properly', we'll want to move this back to hir_ty or into its own crate. So maybe it makes sense to introduce a temporary type here that gets transformed to `ConstScalar` in hir_ty, like `ConstExpr` :thinking:

Reply:
Yeah. I've added some comments around the place. I see what you mean by it being not really ideal that we have any const evaluation in hir_def. I think that we can reuse the lowering code for arrays, but currently I'm unsure if we can get the right state into that spot to be able to do expression evaluation/lowering.